[FAB-18041] Add Node.js to CC as external service (#1509)
hyperledger/fabric: docs/source/cc_service.md

## Writing chaincode to run as an external service

> Currently, the chaincode as an external service model is supported by GO chaincode shim and Node.js chaincode shim.

Reviewer: ```suggestion
Currently, the chaincode as an external service model is supported by Go chaincode shim and Node.js chaincode shim.
```

> ### GO

Reviewer: ```suggestion
### Go
```

> In Fabric v2.0, the GO shim API adds a `ChaincodeServer` type that developers should use to create a chaincode server.  The `Invoke` and `Query` APIs are unaffected. Developers should write to the `shim.ChaincodeServer` API, then build the chaincode and run it in the external environment of choice. Here is a simple sample chaincode program to illustrate the pattern:

Reviewer: ```suggestion
In Fabric v2.0, the Go shim API provides a `ChaincodeServer` type that developers should use to create a chaincode server.  The `Invoke` and `Query` APIs are unaffected. Developers should write to the `shim.ChaincodeServer` API, then build the chaincode and run it in the external environment of choice. Here is a simple sample chaincode program to illustrate the pattern:
```

> Then build the chaincode as suitable to your GO environment.

Reviewer: ```suggestion
Then build the chaincode as suitable to your Go environment.
```

### Node.js

> `fabric-shim` package for Node.js chaincode adds `shim.server` API to run the chaincode as an external service. If you are using contract APIs, you may want to use the `server` command added to `fabric-chaincode-node` CLI to run a contract in the external service mode.

Reviewer: ```suggestion
`fabric-shim` package for Node.js chaincode provides the `shim.server` API to run chaincode as an external service. If you are using contract APIs, you may want to use the `server` command provided by the `fabric-chaincode-node` CLI to run a contract in the external service mode.
```
